Samples/iOS/Objective-C/IOS-SDK-Demo-App/Ad Formats/LONativeAdViewController.m

```diff
@@ -0,0 +1,196 @@
+@property (nonatomic, weak) IBOutlet UIButton *playBtn;
+@property (nonatomic, weak) IBOutlet UIButton *closeBtn;
+
+@property (nonatomic, weak) IBOutlet UITableView *tableView;
+
+
+
+@end
+
+@implementation LONativeAdViewController
+
+- (void)viewDidLoad {
+  [super viewDidLoad];
+  // Do any additional setup after loading the view.
+  
+  self.navigationItem.title = @"Liftoff Native Ad";
+  self.placementLbl.text = self.placementId;
+  
+  self.callbackLogs = [self generateCallbackLogs];
+  
+  self.tableView.delegate = self;
+  self.tableView.dataSource = self;
+}
+
+#pragma mark - IBAction Methods
+
+- (IBAction)btnPressed:(id)sender {
+  if (sender == self.loadBtn && self.placementId != nil) {
+    [self loadNativeAd];
+  } else if (sender == self.playBtn && [self.nativeAd canPlayAd]) {
+    [self playNativeAd];
+  } else if (sender == self.closeBtn) {
+    [self closeNativeAd];
+  }
+}
+
+#pragma mark - Private Methods
+
+- (void)loadNativeAd {
+  // clean up before load
+  if (self.nativeAd != nil) {
+    self.nativeAd.delegate = nil;
+    self.nativeAd = nil;
+    [self resetCallbackLogs];
+  }
+  
+  self.nativeAd = [[VungleNative alloc] initWithPlacementId:self.placementId];
+  self.nativeAd.delegate = self;
+  [self.nativeAd load:nil];
+}
+
+- (void)playNativeAd {
+  [self setUpNativeAdForDisplay];
+  
+  [self.nativeAd registerViewForInteractionWithView:self.nativeAdView
+                                          mediaView:self.mediaView
+                                      iconImageView:self.iconView
+                                     viewController:self
+                                     clickableViews:@[self.iconView,
+                                                      self.downloadBtn,
+                                                      self.nativeAdView]];
+}
+
+- (void)setUpNativeAdForDisplay {
+  self.titleLbl.text = self.nativeAd.title;
+  self.ratingLbl.text = [NSString stringWithFormat:@"Rating: %f", self.nativeAd.adStarRating > 0 ? self.nativeAd.adStarRating : 0];
+  self.sponsorLbl.text = self.nativeAd.sponsoredText;
+  self.adTextLbl.text = self.nativeAd.bodyText;
+  [self.downloadBtn setTitle:self.nativeAd.callToAction forState:UIControlStateNormal];
+  
+  self.nativeAd.adOptionsPosition = NativeAdOptionsPositionTopRight;
+}
+
+- (void)closeNativeAd {
+  [self.nativeAd unregisterView];
+  
+  // reset values inside the labels
+  self.titleLbl.text = @"App Title";
+  self.ratingLbl.text = @"App Rating";
+  self.sponsorLbl.text = @"Sponsored by Text";
+  self.adTextLbl.text = @"Body Text";
+  [self.downloadBtn setTitle:@"CTA Text" forState:UIControlStateNormal];  
+}
+
+- (NSArray *)generateCallbackLogs {
+ return @[
+   [[CallbackLog alloc] initWithTitle:@"nativeAdDidLoad" index:0],
+   [[CallbackLog alloc] initWithTitle:@"nativeAdDidTrackImpression" index:1],
+   [[CallbackLog alloc] initWithTitle:@"nativeAdDidClick" index:2],
+   [[CallbackLog alloc] initWithTitle:@"nativeAdDidFailLoad" index:3],
```

Comment:
Found that `nativeAdDidFailLoad` could change to 
`nativeAdDidFailToLoad`,  otherwise when the delegate `- (void)nativeAdDidFailToLoad:withError:` method is called,  the callback log will not display to be called. Thanks :)

Reply:
@YueVungle thanks for catching this - updating now.